cmd/ike/cmd/install.go

```diff
@@ -0,0 +1,127 @@
+package cmd
+
+import (
+	"github.com/maistra/istio-workspace/pkg/openshift"
+
+	openshiftApi "github.com/openshift/api/template/v1"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+// NewInstallCmd takes care of deploying server-side components of istio-workspace
+func NewInstallCmd() *cobra.Command {
+	installCmd := &cobra.Command{
+		Use:          "install-operator",
+		Short:        "Takes care of deploying server-side components of istio-workspace",
+		SilenceUsage: true,
+		RunE:         installOperator,
+	}
+
+	installCmd.Flags().StringP("namespace", "n", "istio-workspace-operator", "Target namespace to which istio-workspace operator is deployed to.")
+
+	return installCmd
+}
+
+func installOperator(cmd *cobra.Command, args []string) error { //nolint[:unparam]
+	namespace, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+	app, err := newApplier(namespace)
+	if err != nil {
+		return err
+	}
+	if err := app.applyResource("crds/istio_v1alpha1_session_crd.yaml"); err != nil {
```

Comment:
Technically all these could have been replaced with something like:

```go
	resources := []string{"crds/istio_v1alpha1_session_crd.yaml", "service_account.yaml"}
	for _, resource := range resources {
		if r, err := app.applyResource(resource); err != nil { // r = TypeMeta?
			if !errors.IsAlreadyExists(err) {
				log.Error(err, "failed creating " + r)
				return err
			}
			log.Info(r + " already exists")
		}
	}
```

Reply:
as I wrote that I do now see there are applyResource and applyTemplate but.. :)

Reply:
True, I felt this code is crap :) Fixed in 99a78b5